app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt

```diff
@@ -65,12 +71,46 @@ class ProfileListAdapter(
             binding.TxtImg.setText(item.name[0].toString())
             binding.TxtImg.avatarBackgroundColor = bgColors[item.colorIndex]
             binding.timeStamp.text = item.timeInstance
+            binding.pauseSwitch.isChecked = item.pauseSwitch
 
             binding.profileCard.setOnClickListener {
                 val args = Bundle()
                 args.putParcelable("Profile", item)
                 val dialog = DetailsFragment.newInstance(args)
-                dialog.show((parentView.context as FragmentActivity).supportFragmentManager, "DialogFragment")
+                dialog.show(
+                    (parentView.context as FragmentActivity).supportFragmentManager,
+                    "DialogFragment"
+                )
+            }
+            binding.pauseSwitch.setOnClickListener {
+                if (!binding.pauseSwitch.isChecked) {
+                    item.pauseSwitch = false
+                    WorkManager.getInstance(profileViewModel.getApplication())
+                        .cancelAllWorkByTag(item.profileId.toString())
+                    if (StoreSession.readLong(AppConstants.ACTIVE_PROFILE_ID) == item.profileId)
+                        audioManager.ringerMode = StoreSession.readInt(AppConstants.RINGTONE_MODE)
+                } else {
+                    val current: Calendar = Calendar.getInstance()
+                    if (StoreSession.readLong(AppConstants.ACTIVE_PROFILE_ID) == item.profileId) {
+                        if (current.get(Calendar.MINUTE) == 59) {
+                            item.smin = 0
+                            if (current.get(Calendar.HOUR_OF_DAY) == 23)
+                                item.shr = 0
+                            else
+                                item.shr = current.get(Calendar.HOUR_OF_DAY) + 1
+                        } else {
+                            item.smin = current.get(Calendar.MINUTE) + 1
+                            item.shr = current.get(Calendar.HOUR_OF_DAY)
+                        }
+                    }
+
+                    val type by lazy { object : TypeToken<List<Boolean>>() {}.type }
+                    val selectedDays by lazy { Gson() }
+                    item.pauseSwitch = true
+                    val days: MutableList<Boolean> = selectedDays.fromJson(item.d, type)
+                    Utils.setAlarms(parentView.context!!, item, days)
```

Comment:
if parentView.requireContext() exist then use that.

Reply:
@rob729 I checked, but parentView.requireContext() does not exist

Reply:
Okay.